lightly/api/api_workflow_upload_dataset.py

```diff
@@ -0,0 +1,172 @@
+                )
+                success = True
+            except Exception as e:
+                warnings.warn(f"Upload of image {filename} failed with error {e}")
+                success = False
+
+            # update the progress bar
+            tqdm_lock.acquire()  # lock
+            pbar.update(1)  # update
+            tqdm_lock.release()  # unlock
+            # return whether the upload was successful
+            return success
+
+        with ThreadPoolExecutor(max_workers=max_workers) as executor:
+            results = list(executor.map(
+                lambda_, [i for i in range(len(dataset))], chunksize=1))
+
+        if not all(results):
+            msg = 'Warning: Unsuccessful upload(s)! '
+            msg += 'This could cause problems when uploading embeddings.'
+            msg += 'Failed at image: {}'.format(results.index(False))
+            warnings.warn(msg)
+
+        # set image type of data and create initial tag
+        if mode == 'full':
+            img_type = 'full'
+        elif mode == 'thumbnails':
+            img_type = 'thumbnail'
+        else:
+            img_type = 'meta'
+
+        initial_tag_create_request = InitialTagCreateRequest(img_type=img_type)
+        self.tags_api.create_initial_tag_by_dataset_id(body=initial_tag_create_request, dataset_id=self.dataset_id)
+
+    def _upload_single_image(self: ApiWorkflowClient, image, label, filename: str, mode):
+        """Uploads a single image to the Lightly platform.
+
+        """
+
+        # check whether the filename is too long
+        basename = filename
+        if not check_filename(basename):
+            msg = (f'Filename {basename} is longer than the allowed maximum of '
+                   'characters and will be skipped.')
+            warnings.warn(msg)
+            return False
+
+        # calculate metadata, and check if corrupted
+        metadata = check_image(image)
+
+        # generate thumbnail if necessary
+        thumbname = None
+        thumbnail = None
+        if mode == 'thumbnails' and not metadata['is_corrupted']:
+            thumbname = '.'.join(basename.split('.')[:-1]) + '_thumb.webp'
+
+        try:
+            body = SampleCreateRequest(file_name=basename, thumb_name=thumbname, meta_data=metadata)
+            sample_id = self.samples_api.create_sample_by_dataset_id(body=body, dataset_id=self.dataset_id).id
+        except RuntimeError:
+            raise ValueError("Creating the sampling in the web platform failed.")
+
+        if not metadata['is_corrupted'] and mode in ["thumbnails", "full"]:
+            if mode == "thumbnails":
+                is_thumbnail = True
+                thumbnail = get_thumbnail_from_img(image)
+                image_to_upload = PIL_to_bytes(thumbnail, ext='webp', quality=90)
+            else:
+                is_thumbnail = False
+                image_to_upload = PIL_to_bytes(image)
+
+            signed_url = self.samples_api.get_sample_image_write_url_by_id(
```

Comment:
There seems to be a misunderstanding here. We want to upload BOTH thumbnails and the full image in case the mode is "full". 

To break it down:
- mode == "full": upload metadata, thumbnail, image
- mode == "thumbnails": upload metadata, thumbnail
- mode == "meta": upload metadata

Reply:
Changed it in my last commit to do that.